### PR TITLE
added some comments and renamed some variables in PathResolver and PathResolverSnapshot

### PR DESCRIPTION
**Pull request by heejaechang**
this sit on top of https://github.com/microsoft/python-language-server/pull/1596

this is actual change - https://github.com/microsoft/python-language-server/pull/1597/commits/d6c86113d25faa4a7870af182899ac4b2542c0ff - and commit before it is same as #1596

### CODE REVIEW

File: src/Analysis/Ast/Impl/Analyzer/ProgressReporter.cs [hunk 1]
```diff
@@ -20,8 +20,8 @@
 
 namespace Microsoft.Python.Analysis.Analyzer {
     internal sealed class ProgressReporter : IProgressReporter, IDisposable {
-        private const int _initialDelay = 50;
-        private const int _reportingInterval = 100;
+        private readonly static TimeSpan s_initialDelay = TimeSpan.FromMilliseconds(50);
```
Comment (MikhailArkhipov) [inline src/Analysis/Ast/Impl/Analyzer/ProgressReporter.cs]: We don't use Hungarian prefixes

File: src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs [hunk 2]
```diff
@@ -93,7 +94,7 @@ internal sealed class PythonAnalyzerEntry {
         }
 
         public Task<IDocumentAnalysis> GetAnalysisAsync(CancellationToken cancellationToken)
-            => _analysisTcs.Task.ContinueWith(t => t.GetAwaiter().GetResult(), cancellationToken);
+            => _analysisTcs.Task.WaitAsync(cancellationToken);
```
Comment (MikhailArkhipov) [inline src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs]: @AlexanderSher ?

File: src/Analysis/Ast/Impl/Analyzer/ProgressReporter.cs (hunk 1 above)
Comment (heejaechang) [reply src/Analysis/Ast/Impl/Analyzer/ProgressReporter.cs]: so, just _initialDelay?

Comment (jakebailey) [reply src/Analysis/Ast/Impl/Analyzer/ProgressReporter.cs]: In our convention, it'd be `InitialDelay` as it is a static value.

Comment (jakebailey) [reply src/Analysis/Ast/Impl/Analyzer/ProgressReporter.cs]: FWIW I think this change is from #1596, so should be commented there.

File: src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs (hunk 2 above)
Comment (jakebailey) [reply src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs]: Also from #1596, so we should probably review that first.

File: src/LanguageServer/Impl/LanguageServer.Configuration.cs [hunk 3]
```diff
@@ -73,23 +73,23 @@ public sealed partial class LanguageServer {
...
+            settings.diagnosticPublishDelay = GetSetting(analysis, "diagnosticPublishDelay", defaultValue: 1000);
             var ds = _services.GetService<IDiagnosticsService>();
             ds.PublishingDelay = settings.diagnosticPublishDelay;
 
             ds.DiagnosticsSeverityMap = new DiagnosticsSeverityMap(
-                GetSetting(analysis, "errors", Array.Empty<string>()),
-                GetSetting(analysis, "warnings", Array.Empty<string>()),
-                GetSetting(analysis, "information", Array.Empty<string>()),
-                GetSetting(analysis, "disabled", Array.Empty<string>()));
+                GetSetting(analysis, "errors", defaultValue: Array.Empty<string>()),
```
Comment (MikhailArkhipov) [inline src/LanguageServer/Impl/LanguageServer.Configuration.cs]: Not entirely convinced in need to specify defaults explicitly, makes it longer. There are VS extensions that display then if needed.

Comment (heejaechang) [reply src/LanguageServer/Impl/LanguageServer.Configuration.cs]: it makes reviewing code easier though :) especially when constant is used. but not a strong argument. but if team prefer no named argument, I can go and delete those.

**Comment by MikhailArkhipov** [comment]
https://github.com/microsoft/python-language-server/pull/1584 may need to go first

**Comment by heejaechang** [comment]
@MikhailArkhipov ping?

**Comment by MikhailArkhipov** [comment]
Not before #1584 since it touches same area.

**Pull request closed without merging**